proarc-common/src/main/java/cz/cas/lib/proarc/common/object/ndk/NdkMetadataHandler.java

```diff
@@ -574,4 +578,20 @@ public void setMods(ModsDefinition mods) {
         }
     }
 
+    /**
+     * Removes pageType if it is default option - see DEFAULT_PAGE_TYPE
+     *
+     * @param data metadata string
+     * @param isJson set true if metadata are in JSON format, otherwise XML is expected
+     * @return data without default pageType - if was present
+     */
+    private String removeDefaultPageType(String data, Boolean isJson) {
```

Comment:
Typ by se měl rušit jen pro `model:page`. Přece jen bych změnil textový replace a rušil typ v JAXB. Data parsujeme tak jako tak. V případě XML může někdo v budoucnu poslat XML s jinak pojmenovaným namespace než mods a tohle nezafunguje. Pokud je XML v podobě `<mods:part type="NormalPage"/>`, tak by se měl odstranit celý element. Další možná varianta je `<mods:part type="NormalPage"></mods:part>`.

Reply:
Řešení pomocí JAXB jsem zkoušel a byl jsem schopný dosáhnout pouze stavu, kdy se "NormalPage" neuloží a zůstane pouze prázdný element <mods:part /> - ten by se musel smazat transformační šablonou po marshallu, což je ale stejně nepraktické jako moje předchozí řešení pomocí práce s textem.

Jak provést smazání celého elementu v případě, že jeden konkrétní atribut má hodnotu null jsem v rámci samotného JAXB nenašel, pouze EclipseLink MOXy (toto řešení by znamenalo zahození maven-jaxb2-plugin, tedy nepraktické).

Jediná alternativa, která mě momentálně napadá, je vlastní adaptér na celý partDefinition, kde bychom se ale opět dostali do nesnází s parsováním stringu.

Reply:
Dobře. Případ `<mods:part type="NormalPage"/>` by v podstatě neměl nastat, protože správně popsaná strana je očíslovaná. JAXB implementaci bych kvůli tomuto určitě neměnil. Celé mi to přijde zbytečné, protože do současného NDK balíku se ten MODS ani nedostane a NormalPage je povolená hodnota.

Element mods:part by měl jít smazat vyhozením z listu mods.getPart().

Jak už jsem psal minimalizoval bych transformaci jen na patřičný model. Správné místo je nejspíš NdkPageMapper ze kterého dědí OldPrintPageMapper.

Do vlastního parseru bych se nepouštěl. Ta issue za to určitě nestojí.